dotcom-rendering/src/web/browser/bootCmp/init.ts

```diff
@@ -0,0 +1,107 @@
+import { getCookie, log } from '@guardian/libs';
+import { getLocaleCode } from '@frontend/web/lib/getCountryCode';
+import { cmp, onConsentChange } from '@guardian/consent-management-platform';
+import { injectPrivacySettingsLink } from '@root/src/web/lib/injectPrivacySettingsLink';
+import { startup } from '@root/src/web/browser/startup';
+import { ConsentState } from '@guardian/consent-management-platform/dist/types';
+import {
+	OphanComponentType,
+	OphanAction,
+	submitComponentEvent,
+} from '../ophan/ophan';
+
+const trackPerformance = (
+	timingCategory: string,
+	timingVar: any,
+	timingLabel: string,
+): void => {
+	const { ga } = window;
+
+	if (!ga) {
+		return;
+	}
+
+	if (window.performance && window.performance.now) {
+		ga(
+			'allEditorialPropertyTracker.send',
+			'timing',
+			timingCategory,
+			timingVar,
+			Math.round(window.performance.now()),
+			timingLabel,
+		);
+	}
+};
+
+const init = async (): Promise<void> => {
+	/**
+	 * Keep this file in sync with CONSENT_TIMING in static/src/javascripts/boot.js in frontend
+	 * mark: CONSENT_TIMING
+	 */
+	if (!window.guardian.config.switches.consentManagement) return; // CMP turned off!
+	const browserId = getCookie({ name: 'bwid', shouldMemoize: true });
+	const { pageViewId } = window.guardian.config.ophan;
+
+	// Track when CMP will show with GA
+	cmp.willShowPrivacyMessage()
+		.then((willShow) => {
+			trackPerformance(
+				'consent',
+				'acquired',
+				willShow ? 'new' : 'existing',
+			);
+		})
+		.catch((e) =>
+			console.error(`CMP willShowPrivacyMessage - error: ${e}`),
+		);
```

Comment:
FWIW I don’t think anyone is looking at this, sadly. We’ve got better measurement tools anyways, now, but the slowness of CMP is not something we can do anything about at this stage.

Reply:
I'll refactor it to use the new `log` command so it's at least following a standard for messaging